.github/workflows/_matrix-and-codecov-on-merge-to-main.yml

```diff
@@ -29,7 +29,13 @@ jobs:
       run:
         shell: bash -l {0}
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13, macos-14]
+        python-version: ["3.10", "3.11", "3.12"]
+    env:
+          LATEST_PYTHON_VERSION: "3.12"
```

Comment:
What if we create `LATEST_PYTHON_VERSION` at the top? I did a quick search and found no simple way to extract the largest value from the `python-version` array.